lib/applicants/validation.ts

```diff
@@ -23,12 +23,19 @@ export const requestPermitHolderInformationSchema = object({
       then: mixed<Gender>().oneOf(Object.values(Gender)).required('Please select a gender'),
       otherwise: mixed<Gender>().oneOf(Object.values(Gender)).optional(),
     }),
-  otherGender: string().nullable().default(null),
+  otherGender: string().when('type', {
+    is: 'NEW',
+    then: string().nullable().default(null),
+    otherwise: string().optional(),
+  }),
   email: string().email('Please enter a valid email address').nullable().default(null),
   phone: string()
     .required('Please enter a valid phone number')
     .matches(phoneNumberRegex, 'Please enter a valid phone number in the format 000-000-0000'),
-  receiveEmailUpdates: boolean().required(),
+  receiveEmailUpdates: boolean().when('type', {
+    is: 'NEW' || 'RENEWAL',
```

Comment:
Don't think this will work, see https://github.com/jquense/yup#schemawhenkeys-string--string-builder-object--values-any-schema--schema-schema

We would need to use a function here to check that the type is either new or renewal. Please refer to the alternative code in the second example from the link above

Reply:
Fixed